chore(1.10.0): update README and add CHANGELOG (#11852)
longhorn/longhorn: CHANGELOG/CHANGELOG-1.10.0.md

We welcome feedback and contributions to help continuously improve Longhorn.

For terminology and context on Longhorn releases, see [Releases](https://github.com/longhorn/longhorn#releases).

## Removal

### `longhorn.io/v1beta1` API

The `v1beta1` Longhorn API version has been removed.

See [GitHub Issue #10249](https://github.com/longhorn/longhorn/issues/10249) for details.

### `replica.status.evictionRequested` Field

The deprecated `replica.status.evictionRequested` field has been removed.

See [GitHub Issue #7022](https://github.com/longhorn/longhorn/issues/7022) for details.

## Primery Highlights

### New V2 Data Engine Features

#### Interrupt Mode Support

Interrupt mode has been added to the V2 Data Engine to help reduce CPU usage. This feature is especially beneficial for clusters with idle or low I/O workloads, where conserving CPU resources is more important than minimizing latency.

While interrupt mode lowers CPU consumption, it may introduce slightly higher I/O latency compared to polling mode. In addition, the current implementation uses a hybrid approach, which still incurs a minimal, constant CPU load even when interrupts are enabled.

> > **Limitation:** Supports **AIO disks only**.

Reviewer: ```suggestion
> [!NOTE]
> **Limitation:** Supports **AIO disks only**.
```

See [Interrupt Mode](https://longhorn.io/docs/1.10.0/v2-data-engine/features/interrupt-mode) for details.

#### Volume and Snapshot Cloning

> V2 volumes now support cloning.

Reviewer: For each highlighted feature, briefly explain the benefits it offers to users.

For this, suggest briefly explaining the types of clones we support, with emphasis on the benefits of fast/smart-clone.

See [Volume Clone Support](https://longhorn.io/docs/1.10.0/v2-data-engine/features/volume-clone) for details.

#### Replica Rebuild QoS

Provides Quality of Service (QoS) control for V2 volume replica rebuilds. You can configure bandwidth limits globally or per volume to prevent storage throughput overload on source and destination nodes.

See [Replica Rebuild QoS](https://longhorn.io/docs/1.10.0/v2-data-engine/features/replica-rebuild-qos) for details.

#### Volume Expansion

Longhorn now supports volume expansion for V2 Data Engine volumes. You can expand the volume through the UI or by modifying the PVC manifest.

See [V2 Volume Expansion](https://longhorn.io/docs/1.10.0/v2-data-engine/features/volume-expansion) for details.

#### Run Without Hugepage

> The V2 Data Engine can run without Hugepage by setting `data-engine-hugepage-enabled` to `{"v2":"false"}`.

Reviewer: Does this implicitly mean the interruption mode feature? Consider merging them?

Reviewer: compute load for interruption. that's different. clear.

# Longhorn v1.10.0 Release Notes

> Longhorn v1.10.0 delivers a range of new features, enhancements, and bug fixes focusing on improving stability, performance, and overall user experience. Key highlights include improvements to the V2 Data Engine, enhanced resilience, simplified configuration, and better observability.

Reviewer 2: ```suggestion
Longhorn v1.10.0 is a major release focused on improving stability, performance, and the overall user experience. This version introduces significant enhancements to our core features, including the V2 Data Engine, and streamlines configuration for easier management. 

The key highlights include improvements to the V2 Data Engine, enhanced resilience, simplified configuration, and better observability.
```

We welcome feedback and contributions to help continuously improve Longhorn.

For terminology and context on Longhorn releases, see [Releases](https://github.com/longhorn/longhorn#releases).

## Removal

### `longhorn.io/v1beta1` API

The `v1beta1` Longhorn API version has been removed.

See [GitHub Issue #10249](https://github.com/longhorn/longhorn/issues/10249) for details.

### `replica.status.evictionRequested` Field

The deprecated `replica.status.evictionRequested` field has been removed.

See [GitHub Issue #7022](https://github.com/longhorn/longhorn/issues/7022) for details.

> ## Primery Highlights

Reviewer 2: ```suggestion
## Primary Highlights
```

See [Volume Clone Support](https://longhorn.io/docs/1.10.0/v2-data-engine/features/volume-clone) and [GitHub Issue#7794](https://github.com/longhorn/longhorn/issues/7794) for details.

#### Replica Rebuild QoS

Provides Quality of Service (QoS) control for V2 volume replica rebuilds. You can configure bandwidth limits globally or per volume to prevent storage throughput overload on source and destination nodes.

See [Replica Rebuild QoS](https://longhorn.io/docs/1.10.0/v2-data-engine/features/replica-rebuild-qos) and [GitHub Issue#10770](https://github.com/longhorn/longhorn/issues/10770) for details.

#### Volume Expansion

Longhorn now supports volume expansion for V2 Data Engine volumes. You can expand the volume through the UI or by modifying the PVC manifest.

See [V2 Volume Expansion](https://longhorn.io/docs/1.10.0/v2-data-engine/features/volume-expansion) and [GitHub Issue#8022](https://github.com/longhorn/longhorn/issues/8022) for details.

> #### Run Without Hugepage

Reviewer 2: ```suggestion
#### Support for Running Without Hugepages
```

> This reduces memory pressure on low‑spec nodes and increases deployment flexibility. Performance may be lower compared to running with Hugepage.

Reviewer 2: ```suggestion
This reduces memory pressure on low-spec nodes and increases deployment flexibility. Performance may be lower compared to running with Hugepages.
```

See [GitHub Issue#7066](https://github.com/longhorn/longhorn/issues/7066) for details.

### New V1 Data Engine Features

#### IPv6 Support

V1 volumes now support single-stack IPv6 Kubernetes clusters.

> **Warning:** Dual-stack Kubernetes clusters and V2 volumes are **not supported** in this release.

See [GitHub Issue #2259](https://github.com/longhorn/longhorn/issues/2259) for details.

### Consolidated Global Settings

> Longhorn settings are now unified for simpler management across V1 and V2 Data Engines:

Reviewer 2: ```suggestion
To simplify management, Longhorn settings are now unified across V1 and V2 Data Engines, using a new, more flexible JSON format.
```

> ## Longhorn v1.10.0 Release Notes

Reviewer: ```suggestion
# Longhorn v1.10.0 Release Notes
```

See [Issue #2259](https://github.com/longhorn/longhorn/issues/2259) for details.

## V2 Data Engine

### Longhorn System Upgrade

Live upgrades of V2 volumes are **not supported**. Ensure all V2 volumes are detached before upgrading.

### Functional Updates

#### Run Without Hugepage

The V2 Data Engine can run without Hugepage by setting `data-engine-hugepage-enabled` to `{"v2":"false"}`.

This reduces memory pressure on low‑spec nodes and increases deployment flexibility. Performance may be lower compared to running with Hugepage.

#### Interrupt Mode Support

Interrupt mode has been added to the V2 Data Engine to help reduce CPU usage. This feature is especially beneficial for clusters with idle or low I/O workloads, where conserving CPU resources is more important than minimizing latency.

While interrupt mode lowers CPU consumption, it may introduce slightly higher I/O latency compared to polling mode. In addition, the current implementation uses a hybrid approach, which still incurs a minimal, constant CPU load even when interrupts are enabled.

> **Limitation:** Supports **AIO disks only**.

> See [Interrupt Mode](https://longhorn.io/docs/1.10.0/v2-data-engine/features/interrupt-mode) for details.

Reviewer 3: How about

```
[Documentation](https://longhorn.io/docs/1.10.0/v2-data-engine/features/interrupt-mode/) | [GitHub Issue](https://longhorn.io/docs/1.10.0/v2-data-engine/features/interrupt-mode)
```

Author: I’m not sure. If there are multiple documents, it might end up showing as "See <ins>Document</ins> and <ins>Document</ins>", I feel the current approach might be clearer, but I’m happy to change. Could we also get some input from @longhorn/doc ?

Reviewer 3: > Could we also get some input from @longhorn/doc ?

Sure. The format was introduced by @jillian-maroket before. We can check if @sushant-suse has any idea.

Reviewer 2: I agree with @c3y1huang, the current approach seems better. If we have to show both document and issue then we can say like:
```
See [Interrupt Mode](https://longhorn.io/docs/1.10.0/v2-data-engine/features/interrupt-mode/) and [GitHub Issue #12345](https://longhorn.io/issues/12345) for more details.
```

Reviewer 2: I would choose using the `See [Topic] and [GitHub Issue #xxxx] for details.` format as it is more user-friendly option because it is consistent. For example:
- Single Documentation Link: `See [Interrupt Mode] for details.`
- Single GitHub Issue Link: `See [GitHub Issue #xxxx] for details.`
- Both Links: `See [Interrupt Mode] and [GitHub Issue #xxxx] for details.`

So, it is clear, more readable, and flexible.

WDYT?

Author: @derekbit Do you prefer `option 1`?

Reviewer 3: I meant second is complicated
